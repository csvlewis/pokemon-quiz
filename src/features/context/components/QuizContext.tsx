import { createContext } from "react";

import { Option } from "@/types/Option";
import { QuizContextType } from "@/features/context/types/QuizContextType";

export const QuizContext = createContext<QuizContextType>({
  correctAnswers: 0,
  setCorrectAnswers: () => {},
  currentQuestion: 1,
  setCurrentQuestion: () => {},
  displayAnswer: false,
  setDisplayAnswer: () => {},
  questions: [],
  setQuestions: () => {},
  quizInProgress: false,
  setQuizInProgress: () => {},
  quizLength: 5,
  setQuizLength: () => {},
  resetQuiz: () => {},
  selectedAnswer: {} as Option,
  setSelectedAnswer: () => {},
  selectedQuestionTypes: [],
  setSelectedQuestionTypes: () => {},
  showResults: false,
  setShowResults: () => {},
});
