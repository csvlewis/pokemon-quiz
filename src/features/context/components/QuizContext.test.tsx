import React from "react";
import { QuizContext } from "./QuizContext";
import { render } from "@testing-library/react";

describe("QuizContext", () => {
  it("should render children correctly", () => {
    const mockValue = {
      correctAnswers: 0,
      setCorrectAnswers: jest.fn(),
      currentQuestion: 1,
      setCurrentQuestion: jest.fn(),
      displayAnswer: false,
      setDisplayAnswer: jest.fn(),
      questions: [],
      setQuestions: jest.fn(),
      quizInProgress: false,
      setQuizInProgress: jest.fn(),
      quizLength: 5,
      setQuizLength: jest.fn(),
      resetQuiz: jest.fn(),
      selectedAnswer: { text: "", isCorrect: false },
      setSelectedAnswer: jest.fn(),
      showResults: false,
      setShowResults: jest.fn(),
    };

    const { container } = render(
      <QuizContext.Provider value={mockValue}>
        <div>Test</div>
      </QuizContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
