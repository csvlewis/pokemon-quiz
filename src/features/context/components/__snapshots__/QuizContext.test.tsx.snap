// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizContext should render children correctly 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;
